### PR TITLE
Add basic level rating system

### PR DESCRIPTION
**Pull request by vchadha**

### CODE REVIEW

File: Assets/Scripts/Utils/LevelData.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    public LevelData(int starRating)
+    {
+        if (starRating > Constants.MAX_STAR_RATING)
+        {
+            starRating = Constants.MAX_STAR_RATING;
+        }
+        else if (starRating < 0)
+        {
+            starRating = 0;
+        }
```
Comment (justinro-underscore) [inline Assets/Scripts/Utils/LevelData.cs]: Opt: Can use `Mathf.Clamp(starRating, 0, Constants.MAX_STAR_RATING)`!

File: Assets/Scripts/Utils/LevelData.cs
Comment (justinro-underscore) [inline Assets/Scripts/Utils/LevelData.cs:1]: I'm cool with this for now, but it kinda seems to me like this should be inside of the `GameData` class

**Review by justinro-underscore — APPROVED** [review]

**Comment by justinro-underscore** [comment]
💩 🔢

File: Assets/Scripts/Utils/LevelData.cs
Comment (vchadha) [reply Assets/Scripts/Utils/LevelData.cs:1]: oooof totally thought i did that lmao
will fix after obstacles - i have a better idea for how all this should work anyways jeeeesus what was i thinking lmao

Comment (justinro-underscore) [inline Assets/Scripts/Utils/LevelData.cs:1]: This is super minor but I don't think the GameData and LevelData should be in the `Utils/` folder. I would put them in the `Main` folder

File: Assets/Scripts/SceneControllers/LevelSelectController.cs [hunk 2]
```diff
@@ -130,16 +141,12 @@ public bool UnlockNextLevel()
...
     {
-        int lastUnlockedLevelIndex = gameData.lastUnlockedLevelIndex;
-        for (int i = 0; i < levels.Count; i++)
+        for (int i = 0; i < levelData.Count; i++)
         {
-            if (i <= lastUnlockedLevelIndex)
-            {
-                levels[i].UnlockLevel(unlockedIconSprite);
-            }
+            Debug.Log(levelData[i].starRating);
```
Comment (justinro-underscore) [inline Assets/Scripts/SceneControllers/LevelSelectController.cs]: Log statement!

**Review by justinro-underscore — APPROVED** [review]

**Comment by justinro-underscore** [comment]
Note that I may tweak this when I implement the level metrics (like when we assign the level stars)

**Pull request merged**